### ISSUE TITLE
Find() currently returns a slice of pointers. Could it be tweaked to return a slice of plain model values instead?

### ISSUE THREAD

**Issue by preslavrachev** [go-gorm/gen#401]
I noticed that after generation, the `Find` method returns a pair of the following sort:

```go
[]*model.User, error
```

I understand why this could possibly make sense for `First()` (returning a pointer), but the slice itself being a kind of implicit pointer reference, it can be a simple slice of values instead:

```go
[]model.User, error
```

Actually, why do you go returning pointers in the first place, as the user still needs to manually check for `ErrNotFound`. In other words, the `nil`-ability of the end result does not bring much. Or am I missing something?

In short, is there a single option that would tweak methods to return non-pointer values instead?

**Comment by riverchu** [comment]
Sorry for reply late reply.
In GO, parameters are passed between functions using pass-by-value, which means each time you access the value in model array(for range or sth else), the struct will be copied. When the struct array is long, the copy operation can be time-consuming.

**Comment by preslavrachev** [comment]
@riverchu That's true, but I hope you are aware of the fact that a large percentage of Go developers prefer copying of struct values, even when those are part of a slice. You can see this everywhere - even the original GORM documentation encourages this approach everywhere (I have rarely seen a model referred to, or returned as a pointer). In addition, if you look at how people use GORM, nowadays, you will find hundreds, if not thousands of projects wrapping GORM calls in functions like these:

```go
func (r *UserRepository) FindUsers(ids []string) ([]User, error) {
    // GORM call here
}
```

Anyway, I am not trying to discourage your approach, but just give some light on commonly accepted practices among GORM  (and Go) users out there.
